Userland/Libraries/LibGfx/ImageFormats/ILBMLoader.cpp

```diff
@@ -202,6 +224,14 @@ static ErrorOr<void> decode_body_chunk(Chunk body_chunk, ILBMLoadingContext& con
         pixel_data = TRY(planar_to_chunky(body_chunk.data, context));
     }
 
+    // Some files already have 64 colours defined in the palette,
+    // maybe for upward compatibility with 256 colours software/hardware.
+    // DPaint 4 & previous files only have 32 colours so the
+    // palette needs to be extended only for these files.
+    if (context.viewport_mode & ViewportMode::EHB && context.color_table.size() < 64) {
```

Comment:
If the CAMG chunk is always after CMAP you can do that in `decode_cmap_chunk` (and put all palette-related code together)

Reply:
I thought about it but unfortunately there is no guarantee the CMAP chunk comes before the CAMG chunk (and I already found files where it's the opposite).

Reply:
It would have been nice, but unfortunately these old formats often have permissive specifications